app/api/auth.py

```diff
@@ -129,9 +129,35 @@ def login_user(provider, auth_code):
     elif provider == 'google':
         provider_class = GoogleOAuth()
         payload = {
+            'grant_type': 'authorization_code',
             'client_id': provider_class.get_client_id(),
-            'client_secret': provider_class.get_client_secret()
+            'redirect_uri': request.args.get('redirect_uri'),
+            'client_secret': provider_class.get_client_secret(),
+            'code': auth_code + '/' + add_auth_code
         }
+        headers = {"Content-Type": "application/x-www-form-urlencoded"}
+        access_token = requests.post('https://www.googleapis.com/oauth2/v4/token', data=payload, headers=headers).json()
+        id_token = access_token['id_token']
+        user_details = requests.get('https://oauth2.googleapis.com/tokeninfo?id_token=' + id_token).json()
+        if get_count(db.session.query(User).filter_by(google_id=user_details['sub'])) > 0:
+            user = db.session.query(User).filter_by(google_id=user_details['sub']).one()
```

Comment:
Redundant query, just fetch the first user and see if it is none. This is fetching the user twice. First, just to see the count and then again